portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -83,18 +98,68 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     commentEntity.setProperty("commentInput", commentInput);
     commentEntity.setProperty("timestampMillis", timestampMillis);
 
+    // Handle file upload input from form.
+    getUploadedFileUrl(request, "file").ifPresent(fileUrl -> commentEntity.setProperty("fileUrl", fileUrl));
+    
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity); 
 
     // Redirect back to the blog page.
     response.sendRedirect("/blog.html");
   }
 
-  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+  private static String getParameter(HttpServletRequest request, String name, String defaultValue) {
     String value = request.getParameter(name);
     if (value == null) {
       return defaultValue;
     }
     return value;
   }
+
+  /** Returns a URL that points to the uploaded file, or null if the user didn't upload a file. */
+  private static Optional<String> getUploadedFileUrl(HttpServletRequest request, 
```

Comment:
where do you use the `request`?

Reply:
`Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);`
second line in the function

Reply:
oops, sorry had missed that :/ my bad.